### PR TITLE
add new rule to enforce object literal to be used in setResponse

### LINKED ISSUE #64: enforce object literal in setResponse
1. the top level responseContext parameter should be object literal. the following is considered as error:
```
    setResponse(response, responseContext);
```
2. response headers should be object literal. if the service set the response headers referencing a pass-through response's headers directly unintentional headers might be brought in causing confusing errors. e.g. the following is considered as error (disregarding status code):
```
    setResponse(response, {
      status: passThroughResponse.status,
      headers: passThroughResponse.headers,
    });
```
3. if the service set the response body using an Error instance, the firehose data ends up as empty object after structuredClone.
we should detect such usage of `setResponse` method and report error if the value for the error response body is not a plain object literal. e.g.
instead of:
```
    setResponse(response, {
      status: 400,
      body: serverError, // or error as Error
    });
```
it should be done like:
```
    setResponse(response, {
      status: 400,
      body: {
        code: 'INVALID_AT',
        message: 'Invalid at timestamp',
      },
    });
```

[slack reference](https://checkdigit.slack.com/archives/C0EFU0ELC/p1715102390648219)

### PR DESCRIPTION
**Pull request by le-cong**
Fixes #64

### CODE REVIEW

File: src/object-literal-response.ts
Comment (carlansley) [inline src/object-literal-response.ts:1]: Could we make this more generic, i.e. anything that looks like an object literal of form `{ status: number, body: something }`, enforce that `something` is an object literal?

**Review by ramaghanta — APPROVED** [review]
lftm

**Review by jpolavar — APPROVED** [review]
lftm

**Review by carlansley — APPROVED** [review]

**Comment by le-cong** [comment]
@carlansley can you update the github repo protection rule a bit?
`Analyze` has been renamed as `CodeQL / CodeQL analysis` following our standards

**Comment by carlansley** [comment]
> @carlansley can you update the github repo protection rule a bit so that this PR can be squirged? `Analyze` has been renamed as `CodeQL / CodeQL analysis` following our standards

I removed analyze, but for some reason I can't add CodeQL / CodeQL analysis

**Comment by le-cong** [comment]
> > @carlansley can you update the github repo protection rule a bit so that this PR can be squirged? `Analyze` has been renamed as `CodeQL / CodeQL analysis` following our standards
> 
> I removed analyze, but for some reason I can't add CodeQL / CodeQL analysis

maybe it should be just `CodeQL analysis`? `CodeQL / CodeQL analysis` is probably some kinds of presentation/display

**Comment by ramaghanta** [comment]
> > maybe it should be just `CodeQL analysis`? `CodeQL / CodeQL analysis` is probably some kinds of presentation/display
> 
> `CodeQL` is the only option, and it is already required. @ramaghanta have you seen this before? Obviously this is only something we deal with on public repos.

@carlansley, I see "CodeQL" was already added as a required check.

**Comment by carlansley** [comment]
> @carlansley, I see "CodeQL" was already added as a required check.

Yeah the weirdness is `CodeQL / CodeQL analysis (javascript) (pull_request)` is listed as a check, but not required.  And not require-able, apparently.

**Pull request merged**